matio: avoid ac_cv_av_copy check to fix cross building

jwillikers opened the pull request:
Set ac_cv_av_copy to indicate the av_copy function exists.
The function should exist in C99 implementations.

Add passthru tests and update script.
Remove use of pname.
Reduce scope of lib.
Add myself as maintainer.

Fixes #357232

This should be backported to 24.11.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

nim65s commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 357235`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 1 package marked as broken and skipped:</summary>
  <ul>
    <li>btc-rpc-explorer</li>
  </ul>
</details>
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>mathematica</li>
    <li>mathematica-cuda</li>
    <li>mathematica-webdoc</li>
    <li>mathematica-webdoc-cuda</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 107 packages built:</summary>
  <ul>
    <li>aligator</li>
    <li>aligator.doc</li>
    <li>casadi</li>
    <li>chatd</li>
    <li>crocoddyl</li>
    <li>crocoddyl.doc</li>
    <li>faircamp</li>
    <li>gancio</li>
    <li>gnss-sdr</li>
    <li>haskellPackages.gi-vips</li>
    <li>haskellPackages.gi-vips.doc</li>
    <li>imaginary</li>
    <li>imgproxy</li>
    <li>immich</li>
    <li>joplin</li>
    <li>labplot</li>
    <li>lemmy-ui</li>
    <li>mastodon</li>
    <li>matio</li>
    <li>mim-solvers</li>
    <li>netlify-cli</li>
    <li>nip2</li>
    <li>photoprism</li>
    <li>pingvin-share.backend</li>
    <li>pingvin-share.frontend</li>
    <li>pinocchio</li>
    <li>pinocchio.doc</li>
    <li>pleroma</li>
    <li>previewqt</li>
    <li>proxsuite</li>
    <li>proxsuite-nlp</li>
    <li>proxsuite-nlp.doc</li>
    <li>proxsuite.doc</li>
    <li>psitransfer</li>
    <li>python311Packages.aerosandbox</li>
    <li>python311Packages.aerosandbox.dist</li>
    <li>python311Packages.aligator</li>
    <li>python311Packages.aligator.doc</li>
    <li>python311Packages.casadi</li>
    <li>python311Packages.crocoddyl</li>
    <li>python311Packages.crocoddyl.doc</li>
    <li>python311Packages.example-robot-data</li>
    <li>python311Packages.example-robot-data.doc</li>
    <li>python311Packages.mim-solvers</li>
    <li>python311Packages.neuralfoil</li>
    <li>python311Packages.neuralfoil.dist</li>
    <li>python311Packages.pinocchio</li>
    <li>python311Packages.pinocchio.doc</li>
    <li>python311Packages.proxsuite</li>
    <li>python311Packages.proxsuite-nlp</li>
    <li>python311Packages.proxsuite-nlp.doc</li>
    <li>python311Packages.proxsuite.doc</li>
    <li>python311Packages.pyvips</li>
    <li>python311Packages.pyvips.dist</li>
    <li>python311Packages.pyvista</li>
    <li>python311Packages.pyvista.dist</li>
    <li>python311Packages.scooby</li>
    <li>python311Packages.scooby.dist</li>
    <li>python311Packages.sfepy</li>
    <li>python311Packages.sfepy.dist</li>
    <li>python311Packages.tsid</li>
    <li>python311Packages.tsid.doc</li>
    <li>python312Packages.aerosandbox</li>
    <li>python312Packages.aerosandbox.dist</li>
    <li>python312Packages.aligator</li>
    <li>python312Packages.aligator.doc</li>
    <li>python312Packages.casadi</li>
    <li>python312Packages.crocoddyl</li>
    <li>python312Packages.crocoddyl.doc</li>
    <li>python312Packages.example-robot-data</li>
    <li>python312Packages.example-robot-data.doc</li>
    <li>python312Packages.mim-solvers</li>
    <li>python312Packages.neuralfoil</li>
    <li>python312Packages.neuralfoil.dist</li>
    <li>python312Packages.pinocchio</li>
    <li>python312Packages.pinocchio.doc</li>
    <li>python312Packages.proxsuite</li>
    <li>python312Packages.proxsuite-nlp</li>
    <li>python312Packages.proxsuite-nlp.doc</li>
    <li>python312Packages.proxsuite.doc</li>
    <li>python312Packages.pyvips</li>
    <li>python312Packages.pyvips.dist</li>
    <li>python312Packages.pyvista</li>
    <li>python312Packages.pyvista.dist</li>
    <li>python312Packages.scooby</li>
    <li>python312Packages.scooby.dist</li>
    <li>python312Packages.sfepy</li>
    <li>python312Packages.sfepy.dist</li>
    <li>python312Packages.tsid</li>
    <li>python312Packages.tsid.doc</li>
    <li>qq</li>
    <li>rubyPackages.ruby-vips (rubyPackages_3_3.ruby-vips)</li>
    <li>rubyPackages_3_1.ruby-vips</li>
    <li>rubyPackages_3_2.ruby-vips</li>
    <li>rubyPackages_3_4.ruby-vips</li>
    <li>snapweb</li>
    <li>tsid</li>
    <li>tsid.doc</li>
    <li>ueberzugpp</li>
    <li>vips</li>
    <li>vips.bin</li>
    <li>vips.dev</li>
    <li>vips.devdoc</li>
    <li>vips.man</li>
    <li>vipsdisp</li>
    <li>waveterm</li>
    <li>ytfzf</li>
  </ul>
</details>

dotlambda commented on pkgs/by-name/ma/matio/package.nix:
```diff
@@ -1,18 +1,52 @@
     homepage = "http://matio.sourceforge.net/";
-    license = licenses.bsd2;
-    maintainers = [ ];
+    license = with lib.licenses; [ bsd2 ];
```
```suggestion
    license = lib.licenses.bsd2;
```

dotlambda commented on pkgs/by-name/ma/matio/package.nix:
```diff
@@ -1,18 +1,52 @@
   src = fetchurl {
-    url = "mirror://sourceforge/matio/${pname}-${version}.tar.gz";
+    url = "mirror://sourceforge/matio/matio-${version}.tar.gz";
     sha256 = "sha256-CmqgCxjEUStjqNJ5BrB5yMbtQdSyhE96SuWY4Y0i07M=";
```
```suggestion
    hash = "sha256-CmqgCxjEUStjqNJ5BrB5yMbtQdSyhE96SuWY4Y0i07M=";
```

jwillikers replied:
Fixed.

jwillikers committed:
matio: avoid ac_cv_av_copy check to fix cross building

Set ac_cv_av_copy to indicate the av_copy function exists.
The function should exist in C99 implementations.

Add passthru tests and update script.
Remove use of pname.
Reduce scope of lib.
Add myself as maintainer.

dotlambda commented:
Please rebase.
